django/db/models/sql/datastructures.py

```diff
@@ -166,3 +166,25 @@ def equals(self, other, with_filtered_relation):
             self.table_name == other.table_name and
             self.table_alias == other.table_alias
         )
+
+
+class BaseSubquery:
+    join_type = None
+    parent_alias = None
+    filtered_relation = None
+
+    def __init__(self, query, alias):
+        self.query = query
+        assert self.query.subquery is True
```

Comment:
This is probably still a bit WIP, but can we try to use `raise ValueError(...)` instead?